pallets/baby-liminal/src/lib.rs

```diff
@@ -114,40 +155,56 @@ pub mod pallet {
         #[pallet::call_index(0)]
         #[pallet::weight(T::WeightInfo::store_key(key.len() as u32))]
         pub fn store_key(
-            _origin: OriginFor<T>,
+            origin: OriginFor<T>,
             identifier: VerificationKeyIdentifier,
             key: Vec<u8>,
         ) -> DispatchResult {
-            Self::bare_store_key(identifier, key).map_err(|e| e.into())
+            Self::bare_store_key(origin, identifier, key).map_err(|e| e.into())
         }
 
         /// Deletes a key stored under `identifier` in `VerificationKeys` map.
         ///
-        /// Can only be called by a root account.
+        /// Returns the deposit locked. Can only be called by the key owner.
         #[pallet::call_index(1)]
-        #[pallet::weight(T::DbWeight::get().writes(1))]
+        #[pallet::weight(T::WeightInfo::delete_key(T::MaximumVerificationKeyLength::get()))]
         pub fn delete_key(
             origin: OriginFor<T>,
             identifier: VerificationKeyIdentifier,
         ) -> DispatchResult {
-            ensure_root(origin)?;
+            let who = ensure_signed(origin).map_err(|_| Error::<T>::BadOrigin)?;
+            let owner = VerificationKeyOwners::<T>::get(identifier)
+                .ok_or(Error::<T>::UnknownVerificationKeyIdentifier)?;
+
+            ensure!(who == owner, Error::<T>::NotOwner);
+
+            let deposit = VerificationKeyDeposits::<T>::take((&owner, &identifier)).unwrap(); // cannot fail since the key has owner and owner must have made a deposit
```

Comment:
Hurts my eyes that this line is so long due to inlining the comment.

Reply:
I think you should make a pipeline for that

Reply:
Because nothing & nobody else thinks this is a problem

Reply:
Ah... looks like we never put anything about line length into the rustfmt -- I could have a line long of 1000 chars and it would still pass the formatter 🙈